staging/src/k8s.io/apiserver/pkg/util/flowcontrol/watch_tracker.go

```diff
@@ -51,37 +57,90 @@ type ForgetWatchFunc func()
 // of watches in the system for the purpose of estimating the
 // cost of incoming mutating requests.
 type WatchTracker interface {
-	// RegisterWatch reqisters a watch with the provided requestInfo
+	// RegisterWatch reqisters a watch based on the provided http.Request
 	// in the tracker. It returns the function that should be called
 	// to forget the watcher once it is finished.
-	RegisterWatch(requestInfo *request.RequestInfo) ForgetWatchFunc
+	RegisterWatch(r *http.Request) ForgetWatchFunc
 
 	// GetInterestedWatchCount returns the number of watches that are
 	// potentially interested in a request with a given RequestInfo
 	// for the purpose of estimating cost of that request.
 	GetInterestedWatchCount(requestInfo *request.RequestInfo) int
 }
 
+// builtinIndexes represents of set of indexes registered in
+// watchcache that are indexing watches and increase speed of
+// their processing.
```

Comment:
It would be good to include an explanation of what the keys and values are.

Reply:
added